### PR TITLE
Add sonar cloud

### PR DESCRIPTION
**Pull request by johnmay**

### CODE REVIEW

**Comment by javadev** [comment]
Failed to connect to localhost/0:0:0:0:0:0:0:1:9000

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -25,3 +25,32 @@ jobs:
...
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: mvn -B verify org.sonarsource.scanner.maven:sonar-maven-plugin:sonar -Dsonar.projectKey=cdk_cdk
```
Comment (javadev) [inline .github/workflows/maven.yml:56]: These keys need to be added

 -Dsonar.organization=organization -Dsonar.host.url=https://sonarcloud.io

**Comment by johnmay** [comment]
Odd followed the setup instructions and worked before with verify - https://github.com/johnmay/beam/blob/master/.github/workflows/build.yml

**Comment by johnmay** [comment]
Nvm I forgot to add the properties to the pom - will do that in a bit

File: .github/workflows/maven.yml (hunk 1 above)
Comment (javadev) [reply .github/workflows/maven.yml:56]: Code coverge is not configured btw.

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -25,3 +25,32 @@ jobs:
...
+      - name: Cache Maven packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.m2
+          key: ${{ runner.os }}-m2-${{ hashFiles('**/pom.xml') }}
+          restore-keys: ${{ runner.os }}-m2
+      - name: Build and analyze
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}  # Needed to get PR information, if any
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
```
Comment (javadev) [inline .github/workflows/maven.yml:55]: To make it work in forked repositories we need to put sonar token directly without variables.

**Pull request merged**